sys/arm/arm/generic_timer.c

```diff
@@ -292,11 +292,25 @@ cntpct_handler(vm_offset_t va, uint32_t insn, struct trapframe *frame,
 	if ((insn & MRS_MASK) != MRS_VALUE)
 		return (0);
 
-	if (MRS_SPECIAL(insn) != MRS_SPECIAL(CNTPCT_EL0))
+	switch (MRS_SPECIAL(insn)) {
+	case MRS_SPECIAL(CNTPCT_EL0): {
+		int emulate;
+		if (!TUNABLE_INT_FETCH("hw.emulate_phys_counter", &emulate) ||
+		    emulate == 0)
+			return (0);
+		val = READ_SPECIALREG(cntvct_el0);
+		break;
+	}
+#if __has_feature(capabilities)
+	case MRS_SPECIAL(CNTFRQ_EL0):
```

Comment:
Maybe add a comment here that it would normally be accessible but isn't on Morello so we emulate it?

Reply:
Done